src/Mvc/Mvc.Testing/src/WebApplicationFactory.cs

```diff
@@ -136,10 +144,33 @@ internal virtual WebApplicationFactory<TEntryPoint> WithWebHostBuilderCore(Actio
         return factory;
     }
 
-    [MemberNotNull(nameof(_server))]
-    private void EnsureServer()
+    /// <summary>
+    /// Configures the factory to use Kestrel as the server.
+    /// </summary>
+    public void UseKestrel()
     {
-        if (_server != null)
+        if (_server != null || _webHost != null)
+        {
+            throw new InvalidOperationException(Resources.UseKestrelCanBeCalledBeforeInitialization);
+        }
+
+        _useKestrel = true;
+    }
+
+    private static IWebHost CreateKestrelServer(IWebHostBuilder builder)
+    {
+        var host = builder.UseKestrel().Build();
+        host.Start();
+        return host;
+    }
+
+    /// <summary>
+    /// Initializes the instance by configurating the host builder.
+    /// </summary>
+    /// <exception cref="InvalidOperationException">Thrown if the provided <typeparamref name="TEntryPoint"/> type has no factory method.</exception>
+    public void Initialize()
```

Comment:
Why does this need to be public now? It looks like we're calling this internally in the same places we used to call `EnsureServer` and that didn't need to be public.

Reply:
FWIW I had this exact train of thought looking at this last week.